src/vm/checker/identity_pass.rs

```diff
@@ -0,0 +1,29 @@
+use vm::representations::{SymbolicExpression, SymbolicExpressionType};
+use vm::checker::errors::{CheckResult, CheckErrors, CheckError};
+
+fn inner_relabel(args: &mut [SymbolicExpression], index: u64) -> CheckResult<u64> {
+    let mut current = index.checked_add(1)
+        .ok_or(CheckError::new(CheckErrors::TooManyExpressions))?;
+    for expression in &mut args[..] {
+        expression.id = current;
+        current = match expression.expr {
+            SymbolicExpressionType::AtomValue(_) => {
+                current.checked_add(1)
+                    .ok_or(CheckError::new(CheckErrors::TooManyExpressions))
```

Comment:
I see what you're doing here, but do you think `2**64 - 1` too loose of an upper bound for the number of expressions?  Do we want to enforce a smaller number?

Reply:
Not sure about an upper bound on the number of expressions -- on some level, if someone wants to pay to get their giant contract analyzed, that's fine by me. This numbering here (as I think you may have inferred), is only used so that we can "decorate" expressions in the subsequent checker passes by using a map of integers to information.